src/main/java/com/google/api/codegen/transformer/InitCodeTransformer.java

```diff
@@ -669,7 +669,9 @@ private void setInitValueAndComments(
 
       String formatMethodName = "";
       String transformParamFunctionName = "";
-      if (context.getFeatureConfig().useResourceNameConverters(fieldConfig)) {
+      if (context.getFeatureConfig().useResourceNameConvertersInSampleOnly(fieldConfig)
```

Comment:
I think redo this conditional as:
```
if (context.getFeatureConfig().useResourceNameConvertersInSample(fieldConfig)
  && !(context.isFlattenedMethodContext() 
       && context.getFeatureConfig().useResourceNameConverters(fieldConfig)) {
```

This is as clear (or clearer?) and then you can remove both `SampleOnly` methods from feature config, because they are only used here.

Reply:
I feel that it is harder to reason about the boolean logic when expressed that way.

In the end, I'd prefer to do a simplification pass later by adding logic which analyzes source & target types, and automatically adds conversions when necessary. Our current approach of tracking every conversion site is too complex to reason about.

I have broken apart my current boolean expression and added comments so that it is easier to understand.